themes/default/content/blog/deploy-aws-static-website-with-angular/index.md

```diff
@@ -0,0 +1,116 @@
+
+In Angular, the basic building blocks of a website are called _components_. These are reusable structures that contain their own logic and functionality. For example, you could create a button as an Angular component and program it to have a particular look and a particular behavior. Then, you would reuse the button component wherever you need it in the website. The button could also be configurable meaning that, based on different properties, you could change its look and behavior.
+
+A simple website, like the one we’ll be building in this post, is mainly created from a group of Angular components.
+
+## What is the AWS Static Website Component?
+
+The AWS Static Website component from Pulumi makes it easy to deploy an AWS S3 static website and, optionally, add a CloudFront content distribution network (CDN). While you can use any of the programming languages Pulumi supports (TypeScript, JavaScript, Python, Go, .NET, Java, and YAML), we’ll be using TypeScript, since that is what the Angular framework uses.
+
+The great advantage of the AWS Static Website component is that you can have a complete, functioning site in a few minutes. The component encapsulates all the complex logic that an AWS static website requires. Without the component, you would need to spend hours or even days writing and debugging code to get the same result.
+
+If you’re interested in learning more about the component, read [Deploy Static Websites to AWS using 10 lines of YAML]({{< relref "/blog/deploy-aws-static-website-with-yaml" >}}).
+
+## The Angular Components
+
+All the Angular components you need to build the website are available in the GitHub [repository](https://github.com/sean1588/angular-static-site-pulumi). The repository contains:
+
+* A component for the [left navigation bar](https://github.com/sean1588/angular-static-site-pulumi/tree/main/src/app/components/shared/leftnav).
+* A component for the [top navigation bar](https://github.com/sean1588/angular-static-site-pulumi/tree/main/src/app/components/shared/topnav).
+* A component for the [Home](https://github.com/sean1588/angular-static-site-pulumi/tree/main/src/app/components/home) page.
+* A component for the [Menu](https://github.com/sean1588/angular-static-site-pulumi/tree/main/src/app/components/menu) page.
+
+We also use some components that come with the Angular framework.
+
+Each of the components has a TypeScript file that contains the programming logic and an HTML file that describes how the component will render.
+
+## Building the Pulumi Café Website
+
+Here are the instructions to build the website.
+
+1. Clone the GitHub repository located [here](https://github.com/sean1588/angular-static-site-pulumi).
+1. Run `npm install` to install the website’s node dependencies.
```

Comment:
The `npm install` command fails for me with Node 16 or above, apparently because of something having to do with a transitive Python dependency somewhere:

```
npm ERR! gyp verb check python checking for Python executable "python" in the PATH
npm ERR! gyp verb `which` failed Error: not found: python
```

I do have `python`, but it's aliased to `python3` (apparently macOS removed the built-in `python` executable), so I'm not sure what the problem is exactly. Downgrading to Node 14 (with `nvm`) makes things work, but since 16 is active LTS and 18 is current, we should probably assume some number of folks will be a version greater than 14. Not sure what the best solution is, but figured I'd mention it as something to be aware of.

Reply:
ahh hmm... thanks for pointing that out!!  ok going to mess around with this a bit and see if I can find a way around it or see if I get rid of whatever dependency is relying on python. If all else fails, maybe I will add a note about the node version support. But going to try not to go that route as it might be awkward telling the user they need to use a old version of node that is a couple versions behind the current

Reply:
I think I've spent way too much time at this point like 5 hours trying to convert the app over to a version of angular which would be compatible with the node 16. I'm just going to update the readme in the Github repo with the build instructions and say it is only compatible with node 14 as I don't think it is a good use of my time to keep messing around with this as I have already lost an entire day of work fighting with it. I think it is probably better if I just create a very basic angular app with this instead of having a side nav and needing to deal with material and all the dependencies that come with it, but I'll update the github repo and the screenshot in the blog post later when I do that. I think we should just ship this the way it is if that sounds good to you. @cnunciato

Reply:
Sounds good @sean1588, thanks for the valiant effort!

Reply:
wow you are fast! lol. I actually just pushed the commit to take out the build command. so might need to wait a few mins for the preview link to generate.